pkg/download.go

```diff
@@ -75,56 +75,70 @@ func NewCmdDownloadRepository(clientGetter genericclioptions.RESTClientGetter) *
 				return err
 			}
 
-			// get source repository
 			repository, err := client.StashV1alpha1().Repositories(namespace).Get(context.TODO(), repositoryName, metav1.GetOptions{})
 			if err != nil {
 				return err
 			}
-			// unlock local backend
+
 			if repository.Spec.Backend.Local != nil {
 				return fmt.Errorf("can't restore from repository with local backend")
 			}
-			// get repository secret
+
+			// get source repository secret
 			secret, err := kc.CoreV1().Secrets(namespace).Get(context.TODO(), repository.Spec.Backend.StorageSecretName, metav1.GetOptions{})
 			if err != nil {
 				return err
 			}
 
+			tempDir, err := ioutil.TempDir("", "stash-cli")
+			if err != nil {
+				return err
+			}
+			defer os.RemoveAll(tempDir)
+			if err = localDirs.prepareSecretDir(tempDir, secret); err != nil {
+				return err
+			}
+			if err = localDirs.prepareDownloadDir(); err != nil {
+				return err
+			}
+
 			// configure restic wrapper
 			extraOpt := util.ExtraOptions{
-				SecretDir:   docker.SecretDir,
-				EnableCache: false,
-				ScratchDir:  docker.ScratchDir,
+				SecretDir: localDirs.secretDir,
 			}
+			// configure setupOption
 			setupOpt, err := util.SetupOptionsForRepository(*repository, extraOpt)
 			if err != nil {
 				return fmt.Errorf("setup option for repository failed")
 			}
 
-			// write secret and config in a temp dir
-			// cleanup whole tempDir dir at the end
-			tempDir, err := ioutil.TempDir("", "stash-cli")
+			resticWrapper, err := restic.NewResticWrapper(setupOpt)
 			if err != nil {
 				return err
 			}
-			defer os.RemoveAll(tempDir)
 
-			// prepare local dirs
-			if err = localDirs.prepareSecretDir(tempDir, secret); err != nil {
+			localDirs.configDir = filepath.Join(tempDir, configDirName)
+			// dump restic's environments into `restic-env` file.
+			// we will pass this env file to restic docker container.
+			err = resticWrapper.DumpEnv(localDirs.configDir, ResticEnvs)
+			if err != nil {
 				return err
 			}
-			if err = localDirs.prepareConfigDir(tempDir, &setupOpt, &restoreOpt); err != nil {
-				return err
+
+			extraAgrs := []string{
+				"--no-cache",
 			}
-			if err = localDirs.prepareDownloadDir(); err != nil {
-				return err
+
+			// For TLS secured Minio/REST server, specify cert path
+			if _, err := os.Stat(filepath.Join(localDirs.secretDir, restic.CA_CERT_DATA)); err == nil {
+				extraAgrs = append(extraAgrs, "--cacert", filepath.Join(localDirs.secretDir, restic.CA_CERT_DATA))
 			}
 
 			// run restore inside docker
-			if err = runRestoreViaDocker(*localDirs); err != nil {
+			if err = runRestoreViaDocker(*localDirs, extraAgrs, restoreOpt.Snapshots); err != nil {
 				return err
 			}
-			klog.Infof("Repository %s/%s restored in path %s", namespace, repositoryName, restoreOpt.Destination)
+			klog.Infof("Restore completed")
```

Comment:
Make the log more clear. Probably log in the loop where the Snapshot has been restored.